kubectl-plugin/pkg/util/generation/generation.go

```diff
@@ -0,0 +1,87 @@
+package generation
+
+import (
+	"gopkg.in/yaml.v2"
+	corev1 "k8s.io/api/core/v1"
+	corev1ac "k8s.io/client-go/applyconfigurations/core/v1"
+
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	rayv1ac "github.com/ray-project/kuberay/ray-operator/pkg/client/applyconfiguration/ray/v1"
+)
+
+type RayClusterYamlObject struct {
+	ClusterName    string
+	Namespace      string
+	RayVersion     string
+	Image          string
+	HeadCPU        string
+	HeadMemory     string
+	WorkerGrpName  string
+	WorkerCPU      string
+	WorkerMemory   string
+	WorkerReplicas int32
+}
+
+func (rayClusterObject *RayClusterYamlObject) GenerateRayClusterApplyConfig() (*rayv1ac.RayClusterApplyConfiguration, error) {
+	rayClusterApplyConfig := rayv1ac.RayCluster(rayClusterObject.ClusterName, rayClusterObject.Namespace).
+		WithName(rayClusterObject.ClusterName).
+		WithSpec(rayv1ac.RayClusterSpec().
+			WithRayVersion(rayClusterObject.RayVersion).
+			WithHeadGroupSpec(rayv1ac.HeadGroupSpec().
+				WithRayStartParams(map[string]string{"dashboard-host": "0.0.0.0"}).
+				WithTemplate(corev1ac.PodTemplateSpec().
+					WithSpec(corev1ac.PodSpec().
+						WithContainers(corev1ac.Container().
+							WithName("ray-head").
+							WithImage(rayClusterObject.Image).
+							WithResources(corev1ac.ResourceRequirements().
+								WithRequests(corev1.ResourceList{
+									corev1.ResourceCPU:    resource.MustParse(rayClusterObject.HeadCPU),
+									corev1.ResourceMemory: resource.MustParse(rayClusterObject.HeadMemory),
+								}).
+								WithLimits(corev1.ResourceList{
+									corev1.ResourceCPU:    resource.MustParse(rayClusterObject.HeadCPU),
+									corev1.ResourceMemory: resource.MustParse(rayClusterObject.HeadMemory),
+								})).
+							WithPorts(corev1ac.ContainerPort().WithContainerPort(6379).WithName("gcs-server"),
+								corev1ac.ContainerPort().WithContainerPort(8265).WithName("dashboard"),
+								corev1ac.ContainerPort().WithContainerPort(10001).WithName("client")))))).
+			WithWorkerGroupSpecs(rayv1ac.WorkerGroupSpec().
+				WithRayStartParams(map[string]string{"metrics-export-port": "8080"}).
```

Comment:
Is it common to specify this port in the start params?

Reply:
So I would like to leave RayStartParams blank/empty, but currently applyconfigurations will not populate the field if you pass in an empty value: https://github.com/ray-project/kuberay/blob/e9f31556c14fae6391fb27a4a96bfbe01f917d46/ray-operator/pkg/client/applyconfiguration/ray/v1/headgroupspec.go#L54 and RayStartParams is a required field when applying the RayCluster, and the only way I could work around it is by populating the field with default values.

Reply:
yes, when i set it to nil, it won't populate the field. Strings that aren't valid params causes the RayCluster to not start up.

Reply:
> yes, when i set it to nil, it won't populate the field.

What about not nil map? i.e. `rayStartParams: {}`?

Reply:
I tried just using `map[string]string{}` as well, it does not get populated.

Reply:
hmm that's really odd since most examples use it https://github.com/ray-project/kuberay/blob/master/ray-operator/config/samples/ray-cluster.complete.yaml#L88

Reply:
Discussed offline, we will stick with the current work around. I've made a comment to find a better workaround/fix for this.